src/vm_stack.zig

```diff
@@ -289,7 +289,7 @@ const Stack = struct {
 
             const returns_source: []const Val = stack.values[source_begin..source_end];
             const returns_dest: []Val = stack.values[dest_begin..dest_end];
-            std.mem.copy(Val, returns_dest, returns_source);
+            std.mem.copyForwards(Val, returns_dest, returns_source);
```

Comment:
I think this could be `@memcpy` since the stack spaces shouldn't overlap. Though it would be nice to have an assert double-checking that.

Reply:
For this one, `@memcpy` isn't allowed. I've added an if check and copyBackwards because I was running into a failing wasm test where the dest_begin was greater than the source_begin.